packages/package-manager/src/NpmPackageManager.ts

```diff
@@ -0,0 +1,179 @@
+    return 'npm';
+  }
+
+  async installAsync(parameters: string[] = []) {
+    await this._runAsync(['install', ...parameters]);
+  }
+
+  async addGlobalAsync(...names: string[]) {
+    if (!names.length) return this.installAsync();
+    await this._runAsync(['install', '--global', ...names]);
+  }
+
+  async addWithParametersAsync(names: string[], parameters: string[] = []) {
+    if (!names.length) return this.installAsync(parameters);
+
+    const { versioned, unversioned } = this._parseSpecs(names);
+    if (versioned.length) {
+      await this._patchAsync(versioned, 'dependencies');
+      await this.installAsync(parameters);
+    }
+    if (unversioned.length) {
+      await this._runAsync([
+        'install',
+        '--save',
+        ...unversioned.map(spec => spec.raw),
+        ...parameters,
+      ]);
+    }
+  }
+
+  async addAsync(...names: string[]) {
+    await this.addWithParametersAsync(names, []);
+  }
+
+  async addDevAsync(...names: string[]) {
+    if (!names.length) return this.installAsync();
+
+    const { versioned, unversioned } = this._parseSpecs(names);
+    if (versioned.length) {
+      await this._patchAsync(versioned, 'devDependencies');
+      await this._runAsync(['install']);
+    }
+    if (unversioned.length) {
+      await this._runAsync(['install', '--save-dev', ...unversioned.map(spec => spec.raw)]);
+    }
+  }
+
+  async removeAsync(...names: string[]) {
+    await this._runAsync(['uninstall', ...names]);
+  }
+
+  async versionAsync() {
+    const { stdout } = await spawnAsync('npm', ['--version'], { stdio: 'pipe' });
+    return stdout.trim();
+  }
+
+  async getConfigAsync(key: string) {
+    const { stdout } = await spawnAsync('npm', ['config', 'get', key], { stdio: 'pipe' });
+    return stdout.trim();
+  }
+
+  async removeLockfileAsync() {
+    assert(this.options.cwd, 'cwd required for NpmPackageManager.removeLockfileAsync');
+    const lockfilePath = path.join(this.options.cwd, 'package-lock.json');
+    if (fs.existsSync(lockfilePath)) {
+      rimraf.sync(lockfilePath);
+    }
+  }
+
+  async cleanAsync() {
+    assert(this.options.cwd, 'cwd required for NpmPackageManager.cleanAsync');
+    const nodeModulesPath = path.join(this.options.cwd, 'node_modules');
+    if (fs.existsSync(nodeModulesPath)) {
+      rimraf.sync(nodeModulesPath);
+    }
+  }
+
+  // Private
+  private async _runAsync(args: string[]) {
+    if (!this.options.ignoreStdio) {
+      this.log(`> npm ${args.join(' ')}`);
+    }
+
+    // Have spawnAsync consume stdio but we don't actually do anything with it if it's ignored
+    const promise = spawnAsync('npm', [...args], { ...this.options, ignoreStdio: false });
+    if (promise.child.stderr && !this.options.ignoreStdio) {
+      promise.child.stderr
+        .pipe(split(/\r?\n/, (line: string) => line + '\n'))
+        .pipe(new NpmStderrTransform())
+        .pipe(process.stderr);
+    }
+    return promise;
+  }
+
+  private _parseSpecs(names: string[]) {
```

Comment:
we shouldn't use `_` prefix for private functions

Reply:
Fixed in https://github.com/expo/expo/pull/18576/files#diff-ed6baab8aa32f4cba63e7ad6287f12167c9c3b62b771c6574f0e0b77eceeaf7eR89, using only `private` and without the underscore.